### ISSUE TITLE
prop-types rule should understand RelayContainer

### ISSUE THREAD

**Issue by ahfarmer** [yannickcr/eslint-plugin-react#241]
None of the [Relay examples](https://facebook.github.io/relay/) specify propTypes, probably because the propTypes are specified indirectly by the RelayContainer 'fragments'. Explicit specification of propTypes would be duplicate information.

Should relay users simply disable the `prop-types` rule, or can it be made to detect a RelayContainer and use it to validate props?

**Comment by yannickcr** [comment]
I need to investigate on this, since I've never used Relay myself.

But I would accept a PR for this ;)

**Comment by calebmer** [comment]
For some context the following component should not generate errors with the `prop-types` rule:

```js
import React from 'react'
import Relay from 'react-relay'

function User ({ user }) {
  return <div>{user.name}</div>
}

export default Relay.createContainer(User, {
  fragments: {
    user: () => Relay.QL`
      fragment on User {
        name
      }
    `,
  },
})
```

**Comment by maletor** [comment]
I don't think the propTypes info duplicates the Relay fragment. Similar, but still useful to know both.

**Comment by MatthewHerbst** [comment]
Also, writing out the PropTypes is good so that in the future you could strip Relay from a component and still know it's data requirements.

Also, a rule doing this would not be able to tell you if you were getting a string (from Relay) but actually meant to be getting a number. That's something that PropTypes can check by being between the two.

**Comment by ljharb** [comment]
I think that handling special cases of "not React components" is out of scope of this plugin, and "always providing propTypes" is a strong best practice.
